src/libstore/globals.hh

```diff
@@ -313,6 +313,22 @@ public:
           `uid-range` system feature.
         )"
     };
+
+    Setting<uint64_t> memoryHigh{
+        this, 0, "memory-high",
+        R"(
+          sets the cgroup memory.high limit, causing the build to throttle if exceeded
+          depends on use-cgroups=true
+        )"
+    };
+
+    Setting<uint64_t> memoryMax{
+        this, 0, "memory-max",
+        R"(
+          sets the cgroup memory.max limit, causing the build to be killed if exceeded
+          depends on use-cgroups=true
```

Comment:
I am missing some documentation here in what relation to each other you should set memory-high and memory-max. I would assume something like memory-high 3GB and memory-max 4GB?

Reply:
https://facebookmicrosites.github.io/cgroup2/docs/memory-controller.html

Reply:
depends on how much a specific derivation needs, the goal is to keep track of how much memory each derivation is using, and not schedule too many on one machine
then use cgroups to enforce that it stays within the reserved amounts